### ISSUE TITLE
REVDEPs failures from 0.19.x

### ISSUE THREAD

**Issue by avsm** [mirage/ocaml-cohttp#418]
websocket fails:
http://opam.ocaml.org/builds/f143251e7965d77295234e1f2197e701bd7f021d/logs/local-ubuntu-14.04-ocaml-4.02.1/raw/websocket.html#bottom

```
# File "lib/websocket_lwt.ml", line 30, characters 6-22:
# Error: Unbound value CU.Request.write
```

message-switch fails with missing `iter`:

```
# + /home/opam/.opam/system/bin/ocamlfind ocamlc -c -g -annot -bin-annot -I core -package async -package cohttp -package cohttp.async -package cohttp.lwt -package lwt -package lwt.unix -package re -package re.str -package rpclib -package rpclib.json -syntax camlp4o -package rpclib.syntax -package threads -package unix -thread -I core -o core/protocol.cmo core/protocol.ml
# File "core/protocol.ml", line 484, characters 7-11:
# Error: Unbound value iter
```

**Comment by djs55** [comment]
Hmm, it's probably bad that message-switch has `include Cohttp.S.IO` and then depends on the result including functions like `iter` and `map`. Thanks for adding the upper-bounds. I've also patched this in [djs55/message-switch#33]

**Comment by rgrinberg** [comment]
I think this is now resolved. Can be re-opened if not.
